### DISCUSSION TITLE
compare two columns in a aggregate function using fieldReference

### DISCUSSION THREAD

**Discussion by eric-full-stack** [prisma/prisma#16726] [Q&A]
Hi everyone,
This is my query. I want to take tasks where the column `workedHours` is lte `hours`.
I am using fieldReference in prisma

```
 generator client {
   provider        = "prisma-client-js"
   previewFeatures = ["referentialIntegrity", "fieldReference"]
}
```

and this is how i'm doing
```
    await ctx.prisma.task.aggregate({
          where: {
            user: {
              id: input.id,
            },
           workedHours: {
             lte: ctx.prisma.task.fields["hours"],
           },
         },
         _sum: {
            hours: true,
            workedHours: true,
         },
      });
```

when i run the code i get this error message:
`TypeError: Cannot read properties of undefined (reading 'hours')`
and when I `console.log(ctx.prisma.task)`, i got this
```
{
  findUnique: [Function (anonymous)],
  findUniqueOrThrow: [Function (anonymous)],
  findFirst: [Function (anonymous)],
  findFirstOrThrow: [Function (anonymous)],
  findMany: [Function (anonymous)],
  create: [Function (anonymous)],
  createMany: [Function (anonymous)],
  delete: [Function (anonymous)],
  update: [Function (anonymous)],
  deleteMany: [Function (anonymous)],
  updateMany: [Function (anonymous)],
  upsert: [Function (anonymous)],
  aggregate: [Function (anonymous)],
  groupBy: [Function (anonymous)],
  count: [Function (anonymous)]
}
```
with no fields property.. but when i open TS files of prisma there is the fields property there.. look

```
TaskDelegate {
 ...
    /**
     * Fields of the Task model
     */
    readonly fields: TaskFieldRefs;
}
```
**PRISMA CLIENT VERSION `"@prisma/client": "^4.6.1"`**

what I'm doing wrong?

Thx

**Comment by nurul3101** [comment]
Hey @eric-full-stack 👋 
Can you please provide your `task` model so that I can run it locally and investigate what could be the issue?

I tried to run it with this sample Task model and it works as expected for me:
```prisma
model Task {
  id          Int      @id @default(autoincrement())
  title       String
  hours       Int
  workedHours Int
  completed   Boolean  @default(false)
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
}
```
<img width="532" alt="Screenshot 2022-12-13 at 7 34 24 PM" src="https://user-images.githubusercontent.com/13049676/207353951-5a85e515-eabc-4412-a3b5-eae4dbf641cf.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
